config/v1alpha1/types_insights.go

```diff
@@ -57,8 +57,38 @@ type GatherConfig struct {
 	// Run the following command to get the names of last active gatherers:
 	// "oc get insightsoperators.operator.openshift.io cluster -o json | jq '.status.gatherStatus.gatherers[].name'"
 	// An example of disabling gatherers looks like this: `disabledGatherers: ["clusterconfig/machine_configs", "workloads/workload_info"]`
+	// +kubebuilder:validation:MaxItems=100
+	// +kubebuilder:validation:items:MaxLength=256
 	// +optional
 	DisabledGatherers []string `json:"disabledGatherers"`
+	// storageSpec is an optional field that allows user to define persistent storage for on-demand gathering
+	// jobs to store the Insights data archive.
+	// If omitted, the gathering job will use ephemeral storage.
+	// +optional
+	StorageSpec *StorageSpec `json:"storageSpec,omitempty"`
+}
+
+type StorageSpec struct {
+	// persistentVolumeClaimName is required field that specifies the name of the PersistentVolumeClaim that will
+	// be used to store the Insights data archive. The persistenVolumeClaim must be created in the openshift-insights namespace.
+	// +required
+	PersistentVolumeClaimName PersistentVolumeClaimReference `json:"persistentVolumeClaimName"`
```

Comment:
This might be better as `PersistentVolumeClaim` to reduce repetition of the word "name" when used like:
```yaml
...
persistentVolumeClaimName:
  name: foo 
...
```

Reply:
Done